api/src/client/DynamicsLicenseApplicationIdClient.test.ts

```diff
@@ -0,0 +1,171 @@
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+        {
+          rgb_appnumber: "MWLI00030627",
+          rgb_number: "39PC00105700",
+          rgb_startdate: "2023-06-08T05:09:14Z",
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: "2023-09-30T00:00:00Z",
+          statecode: 0,
+          statuscode: 100000006,
+          rgb_applicationid: "6cbcd136-ba05-ee11-a81c-001dd80648b3",
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+        {
+          rgb_appnumber: "MWLI00030631",
+          rgb_number: "39PC00105800",
+          rgb_startdate: "2023-06-08T05:12:52Z",
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: "2023-09-30T00:00:00Z",
+          statecode: 0,
+          statuscode: 100000006,
+          rgb_applicationid: "d0bfe3a2-ba05-ee11-a81c-001dd80648b3",
+          _rgb_businessid_value: "6182f96b-2105-ee11-a81c-001dd805ef6c",
+        },
+      ],
+    };
+
+    mockAxios.get.mockResolvedValue({ data: { ...mockLicenseApplicationIdResponse } });
+    expect(
+      await client.getLicenseApplicationId(mockAccessToken, mockBusinessId, "Public Movers and Warehousemen")
+    ).toEqual({
+      expirationDate: "2023-09-30T00:00:00Z",
+      applicationId: "7b5c178c-2805-ee11-a81c-001dd805ef6c",
+      licenseStatus: "ACTIVE",
+    });
+    expect(mockAxios.get).toHaveBeenCalledWith(
+      `${ORG_URL}/api/data/v9.2/rgb_applications?$select=rgb_appnumber,rgb_number,rgb_startdate,rgb_versioncode,rgb_expirationdate,statecode,statuscode,rgb_applicationid&$filter=(_rgb_businessid_value eq ${mockBusinessId} and _rgb_apptypeid_value eq ${publicMoversAndWarehousemenLicenseTypeId})&$top=50`,
+      {
+        headers: {
+          Authorization: `Bearer ${mockAccessToken}`,
+        },
+      }
+    );
+  });
+
+  it("only returns active applications based on statecode", async () => {
+    const mockLicenseApplicationIdResponse = {
+      value: [
+        {
+          rgb_appnumber: "TELI00030564",
+          rgb_number: null,
+          rgb_startdate: null,
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: null,
+          statecode: 0,
+          statuscode: 100000005,
+          rgb_applicationid: "license-id-1357",
+        },
+        {
+          rgb_appnumber: "INACTIVE1234",
+          rgb_number: null,
+          rgb_startdate: null,
+          rgb_versioncode: 100000000,
+          rgb_expirationdate: null,
+          statecode: 1,
+          statuscode: 100000005,
+          rgb_applicationid: "license-id-1357",
+        },
+      ],
+    };
+
+    mockAxios.get.mockResolvedValue({ data: { ...mockLicenseApplicationIdResponse } });
+    expect(
+      await client.getLicenseApplicationId(mockAccessToken, mockBusinessId, "Public Movers and Warehousemen")
+    ).toEqual({
+      expirationDate: "",
+      applicationId: "license-id-1357",
+      licenseStatus: "APPROVED",
```

Comment:
[pebble] I think in order for this test to really give confidence, the Inactive record should come first in the mocked value.

Right now, we're just selecting and returning the `[0]` array element. If you removed the logic that filters out inactive applications, this test would still pass because it'd just return the first element from the array.

If you switch the order in the mock response here, then the test would only pass if the filtering really is happening.